{{ cookiecutter.__package_name_kebab_case }}/Dockerfile

```diff
@@ -39,6 +31,27 @@ ENV VIRTUAL_ENV /opt/{{ cookiecutter.__package_name_kebab_case }}-env
 # Set the working directory.
 WORKDIR /workspaces/{{ cookiecutter.__package_name_kebab_case }}/
 
+FROM base as builder
+
+USER root
+
+# Install Poetry in separate venv so it doesn't pollute the main venv.
+ENV POETRY_VERSION 1.6.1
+ENV POETRY_VIRTUAL_ENV /opt/poetry-env
+RUN --mount=type=cache,target=/root/.cache/pip/ \
+    python3 -m venv $POETRY_VIRTUAL_ENV && \
+    $POETRY_VIRTUAL_ENV/bin/pip install -U pip setuptools && \
+    $POETRY_VIRTUAL_ENV/bin/pip install poetry~=$POETRY_VERSION
+RUN ln -s $POETRY_VIRTUAL_ENV/bin/poetry /usr/bin/poetry
+
+# Install compilers that may be required for certain packages or platforms.
+RUN --mount=type=cache,target=/var/cache/apt/ \
+    --mount=type=cache,target=/var/lib/apt/ \
+    apt-get update && \
+    apt-get install --no-install-recommends --yes build-essential
+
+USER user
```

Comment:
I think the risk of things going wrong will be less if the venv is installed by the non-root user. Otherwise, we could run into permission issues later on when running the image with the non-root user.

This suggestion moves the `USER root` command down the apt-get install section.
```suggestion
# Install Poetry in separate venv so it doesn't pollute the main venv.
ENV POETRY_VERSION 1.6.1
ENV POETRY_VIRTUAL_ENV /opt/poetry-env
RUN --mount=type=cache,target=/root/.cache/pip/ \
    python3 -m venv $POETRY_VIRTUAL_ENV && \
    $POETRY_VIRTUAL_ENV/bin/pip install -U pip setuptools && \
    $POETRY_VIRTUAL_ENV/bin/pip install poetry~=$POETRY_VERSION
RUN ln -s $POETRY_VIRTUAL_ENV/bin/poetry /usr/bin/poetry

# Install compilers that may be required for certain packages or platforms.
USER root
RUN --mount=type=cache,target=/var/cache/apt/ \
    --mount=type=cache,target=/var/lib/apt/ \
    apt-get update && \
    apt-get install --no-install-recommends --yes build-essential
USER user
```

Reply:
Ignore this comment, I missed something important when I posted it.